src/CtaLink/CtaLink.js

```diff
@@ -15,12 +15,29 @@ import { SrOnlyText } from "../SrOnlyText/SrOnlyText";
  */
 export const CtaLink = React.forwardRef(
   (
-    { className, text, srText, color, icon, iconProps, animate, ...props },
+    {
+      className,
+      element,
+      text,
+      srText,
+      color,
+      icon,
+      iconProps,
+      animate,
+      ...props
+    },
     ref
   ) => {
     // Defaults & Variables.
     // ---------------------------------------------------------------------------
     const levers = {};
+    let Element = "a";
+
+    // Determine whether a or element tag should be used based on children props
+    // ---------------------------------------------------------------------------
+    if (element) {
+      Element = { element };
+    }
```

Comment:
Instead of changing the element, when a child element is present, use what was passed in as the root element returned but modifying it first by cloning it with your added config (props)

Re: https://reactjs.org/docs/react-api.html#cloneelement

```
... 
if (children) {
  let Element = React.cloneElement(children, {your props here})
}

Reply:
Some experimenting with this will be required around the children values.

Reply:
Did a few codesandbox tests to understand how `React.cloneElement()` works then tried implementing it within the CtaLink. Mmm, I don't believe we can use the method in a functional component. We'll need to rewrite the CtaLink as a class component which could conflict with `React.forwardRef`. 🤔  We might need to go back to the drawing board for an alternative solution.

Just to confirm — our end goal here is to be able to pass a component (i.e. Gatsby <Link>, StoryBlok <SbLink>) into our CtaLink component so that the Icon would render within the following html structure:
 
```
<a href="/">
     Cta Link <svg /> 
     // where the <svg /> is the Icon from the <Icon /> component
</a>
```
instead of the implementation written in #58, where it would render:

```
<span>
     <a href="/">
          Cta Link
     </a>
     <svg /> 
     // where the <svg /> is the Icon from the <Icon /> component
</span>
``` 
right? 😃

Reply:
I'll let Shea answer the implementation question, but want to come in here and say that I like your first html structure better. It's more semantic, and works better with TW classes, e.g., `group-hover`, `group-focus`. For the 2nd one, to style the icon focus states, we'll have to use a `group-focus-within` class, which would require us to turn on more variants since `group-focus-within` isn't generated by default decanter core v7 😄  In either case, we'll need to be able to pass classnames and ref to the link also.

Reply:
@sherakama a penny for your thoughts on the implementation and what we're trying to achieve on the HTML structure side of things for the CTA Links? 🤔 

I'll revisit this in the upcoming sprint to see if there's another approach we can take as well.